src/watcher.lisp

```diff
@@ -69,11 +70,15 @@
                         (progn
                           (log-info :logger logger "Hierarchy watcher started")
                           (loop for *watcher-event-time* = (get-universal-time)
-                                do (hierarchy-watcher-once)
-                                until (bt-sem:wait-on-semaphore *stop-semaphore* :timeout *hierarchy-watcher-heartbeat*)))
+                             do (hierarchy-watcher-once)
+                             until (handler-case
+                                       (bt:with-timeout (*hierarchy-watcher-heartbeat*)
```

Comment:
``with-timeout`` used like this can perform async unwinds which are unsafe. I would much prefer using a condition variable with ``(condition-wait … :timeout …)`` on implementations that support it and ``(with-timeout (..) (condition-wait …))`` where it is unavoidable.